src/main/scala/workflow/Gather.scala

```diff
@@ -0,0 +1,39 @@
+package workflow
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+private[workflow] class GatherTransformer[T] extends TransformerNode[Seq[T]] {
+  def transform(dataDependencies: Seq[_], fitDependencies: Seq[TransformerNode[_]]): Seq[T] = dataDependencies.map(_.asInstanceOf[T])
+
+  def transformRDD(dataDependencies: Seq[RDD[_]], fitDependencies: Seq[TransformerNode[_]]): RDD[Seq[T]] = {
+    dataDependencies.map(_.asInstanceOf[RDD[T]].map(t => Seq(t))).reduceLeft((x, y) => {
+      x.zip(y).map(z => z._1 ++ z._2)
```

Comment:
I'm not sure. It looks like spark has zipPartitions up to 4 rdds at once, so we could do a little bit better than one job per RDD. With the way the DAG execution graph is being represented here though we can't really pick the alternate execution plans of the preceding nodes (meaning we apply each of the previous paths' single apply and do it all on a single RDD).